src/test/regress/sql/single_node.sql

```diff
@@ -409,6 +409,45 @@ SELECT count(*) FROM collections_list_1 WHERE key = 11;
 ALTER TABLE collections_list DROP COLUMN ts;
 SELECT * FROM collections_list, collections_list_0 WHERE collections_list.key=collections_list_0.key  ORDER BY 1 DESC,2 DESC,3 DESC,4 DESC LIMIT 1;
 
+-- test hash distribution using INSERT with generate_series() function
+CREATE OR REPLACE FUNCTION part_hashint4_noop(value int4, seed int8)
+RETURNS int8 AS $$
+SELECT value + seed;
+$$ LANGUAGE SQL IMMUTABLE;
+
+CREATE OPERATOR CLASS part_test_int4_ops
+FOR TYPE int4
+USING HASH AS
+operator 1 =,
+function 2 part_hashint4_noop(int4, int8);
+
+CREATE TABLE hash_parted (
+	a int,
+  b int
+) PARTITION BY HASH (a part_test_int4_ops);
+CREATE TABLE hpart0 PARTITION OF hash_parted FOR VALUES WITH (modulus 4, remainder 0);
+CREATE TABLE hpart1 PARTITION OF hash_parted FOR VALUES WITH (modulus 4, remainder 1);
+CREATE TABLE hpart2 PARTITION OF hash_parted FOR VALUES WITH (modulus 4, remainder 2);
+CREATE TABLE hpart3 PARTITION OF hash_parted FOR VALUES WITH (modulus 4, remainder 3);
+
+SELECT create_distributed_table('hash_parted ', 'a');
+
+INSERT INTO hash_parted VALUES (1, generate_series(1,10));
+
+SELECT * FROM hash_parted;
+
+ALTER TABLE hash_parted DETACH PARTITION hpart0;
+ALTER TABLE hash_parted DETACH PARTITION hpart1;
+ALTER TABLE hash_parted DETACH PARTITION hpart2;
+ALTER TABLE hash_parted DETACH PARTITION hpart3;
+
+-- test range partition without creating partitions and inserting with generate_series()
+-- should error out even in plain PG since no partition of relation "parent_tab" is found for row
+-- in Citus it errors out because it fails to evaluate partition key in insert
+CREATE TABLE parent_tab (id int) PARTITION BY RANGE (id);
+SELECT create_distributed_table('parent_tab', 'id');
+INSERT INTO parent_tab VALUES (generate_series(0, 29));
+
```

Comment:
some tests with distributed tables:

```sql
create table table1 (a int);
select create_distributed_table('table1','a');

-- test with returning
with wcte as (insert into table1 values (42) returning *)
insert into table2 values ('hello world');

-- even if we do multi-row insert, it is not fast path router due to cte
with wcte as (select 1 as col)
insert into table2 values ('hello world'), ('hello world');
```

Reply:
Also, it might be better to check the rows inserted for all the tests to make sure that we actually insert the data as expected.
Before doing that, maybe we can insert less rows (like VALUES `(generate_series(0, 3)`) to make tests less verbose

Reply:
Good idea, I will add them at `with_modifying.sql` file